axon-server-connector/src/main/java/org/axonframework/axonserver/connector/command/AxonServerCommandBus.java

```diff
@@ -325,9 +337,11 @@ public void onCompleted() {
                                                   ErrorCode.COMMAND_DISPATCH_ERROR.convert(errorMessage))
                                           );
                                       }
+                                      shutdownLatch.decrement();
                                   }
                               }
                     );
+            shutdownLatch.increment();
```

Comment:
Already was in doubt about the method naming; personally I was playing with "operation" rather than "activity". Renamed the methods to `registerOperation` and `deregisterOperation`.

And good call on the point in time when calling `increment`/`registerOperation`. Moved this to before `try` statement. Also added deregister in the catch block.